### DISCUSSION TITLE
The failure of fluid calculation

### DISCUSSION THREAD

**Discussion by Submarinewake** [idaholab/moose#22336] [Q&A Modules: Navier-Stokes] [answered]
Hello experts,
I recently tried to calculate a complex channel with MOOSE，calculate its temperature conduction and flow field velocity and pressure. But the results obtained are not ideal. I found that the temperature can be calculated smoothly, but the fluid seems not to be calculated. In the results of the flow field, both pressure and velocity are approximately to 0, not even the inlet velocity. After calculation, the model has a low Reynolds number, and theoretically it can be calculated with a laminar model. Do you have any good suggestions？Sincerely look forward to your reply.
`![image](https://user-images.githubusercontent.com/115382145/194734182-01f9bb68-748a-46cc-8bd6-21828675148a.png)
![image](https://user-images.githubusercontent.com/115382145/194734355-964e080a-fd49-480d-9f10-c38db6af6c32.png)

My input file as follow

```
fluid_mu = 0.001   # Pa*s
fluid_rho = 998.2    #kg/m3
fluid_cp = 4128      #j/kg-k
fluid_k = 0.6     #w/m-k
plate_k = 30
source_k = 401


[Mesh]
  type = FileMesh #Read in mesh from file
  file = htcc1_2.e
[]


[Variables]
  [Tf]
    order = FIRST
    family = LAGRANGE
    block = 'fluid'
    initial_condition = 293
  []
  [Ts]
    order = FIRST
    family = LAGRANGE
    block = 'plate source'
    initial_condition = 293
  []
  [./velocity]
    family = LAGRANGE_VEC
    block='fluid'
  [../]
  [./p]
    order = FIRST
    family = LAGRANGE
    block='fluid'
  [../]
[]

[ICs]
  [velocity]
    type = VectorConstantIC
    x_value = 1e-15
    y_value = 2.87
    z_value = 1e-15
    variable = velocity
  []
[]


[Kernels]
 [./mass]
    type = INSADMass
    variable = p
  [../]
  [./mass_pspg]
    type = INSADMassPSPG
    variable = p
  [../]
  [./momentum_convection]
    type = INSADMomentumAdvection
    variable = velocity
  [../]
  [./momentum_viscous]
    type = INSADMomentumViscous
    variable = velocity
  [../]
  [./momentum_pressure]
    type = INSADMomentumPressure
    variable = velocity
    pressure = p
    integrate_p_by_parts = true
  [../]
  [./momentum_supg]
    type = INSADMomentumSUPG
    variable = velocity
    velocity = velocity
  [../]
  [./temperature_advection]
    type = INSADEnergyAdvection
    variable = Tf
    block = 'fluid'
  [../]
  [fluid_temperature_conduction]
    type = ADHeatConduction
    variable = Tf
    block = 'fluid'
    thermal_conductivity = 'k'
  []
  [temperature_supg]
    type = INSADEnergySUPG
    variable = Tf
    velocity = velocity
    block = 'fluid'
  []

  [solid_temperature_conduction]
    type = ADHeatConduction
    variable = Ts
    block ='plate source'
    thermal_conductivity = 'k'
  []
[]

[InterfaceKernels]
  [./convection_heat_transfer]
    type = ConjugateHeatTransfer
    variable = Tf
    T_fluid = Tf
    neighbor_var = 'Ts'
    boundary = 'interface'
    htc = 50000
  [../]
[]

[BCs]
  [./plane_wall]#PCB
    type = ADConvectiveHeatFluxBC
    variable = Ts
    boundary = 'wall'
    T_infinity = 293
    heat_transfer_coefficient = 25
  [../]
  [./no_slip]
    type = VectorFunctionDirichletBC
    variable = velocity
    boundary = 'interface'
  [../]
  [./vec_inlet]
    type = VectorFunctionDirichletBC
    variable = velocity
    boundary = 'inlet'
    function_y = 2.87 #m/s
  [../]
  [./inlet_temp]
    type = DirichletBC
    variable =Tf
    boundary = 'inlet'
    value = 303
  [../]
  [./outlet_p]
    type = DirichletBC
    variable = p
    boundary = 'outlet'
    value = 0
  [../]
  [./source]
    type = NeumannBC
    variable = Ts
    boundary = 'flux'
    value = 2.432e5
  [../]
[]

[Materials]
  [./plate_mat]
    type = ADGenericConstantMaterial
    prop_names = 'k'
    prop_values = '${plate_k}'
    block = 'plate'
  [../]
  [./source_mat]
    type = ADGenericConstantMaterial
    prop_names = 'k'
    prop_values = '${source_k}'
    block = 'source'
  [../]
  [./fluid_mat]
    type = ADGenericConstantMaterial
    prop_names = 'rho mu cp k'
    prop_values = '${fluid_rho} ${fluid_mu}  ${fluid_cp}  ${fluid_k}'
    block = 'fluid'
  [../]
  [ins_mat]
    type = INSADStabilized3Eqn
    velocity = velocity
    pressure = p
    temperature = Tf
    block='fluid'
  []
[]
[Preconditioning]
  [./SMP]
    type = SMP
    full = false
    petsc_options_iname = '-pc_type -pc_factor_mat_solver_package'
    petsc_options_value = ' lu       mumps'
    solve_type = PJFNK
  [../]
[]

[Executioner]
  type = Steady
  l_max_its = 20
  nl_max_its =1000
 
  l_tol = 1e-5
  nl_rel_tol = 1e-3

  solve_type = PJFNK
  ##petsc_options_iname = '-pc_type -pc_hypre_type'
  ##petsc_options_value = 'hypre boomeramg'
   petsc_options_iname = '-pc_type -pc_factor_mat_solver_package -ksp_gmres_restart'
  petsc_options_value = 'lu       superlu_dist                  200'

[]
[Debug]
  show_var_residual_norms = true
[]

[Outputs]
  exodus = true
  file_base = htccflux_out1
  perf_graph = true
[]
```

**Comment by maxnezdyur** [comment, marked as answer]
IIRC when using `integrate_by_parts = true` you shouldn't apply a DirchetBC on pressure (or if you do it requires the tangential velocity to be 0). Run the input file without the pressure outlet bc and if your computational domain is large enough, then the pressure near the outlet will still be close to 0, which may or may not be good enough for your simulation.

**Reply by Submarinewake** [reply]
Thanks for your answer, I will try your suggestion!

**Comment by GiudGiud** [comment]
Hello

Since the flow is mainly driven by the inlet condition, I would check the mesh and in particular the inlet boundary to see that it's as expected.

Guillaume

**Reply by Submarinewake** [reply]
Hello Guillaume
Actually I think it is correlated with the grid, when I modified the input file and made a mesh refinement, my results will not converge, according to the convergence residuals, mainly the residuals of the fluid Large. From the experience in simulation, Fluid simulations have high mesh requirements.
But in the process of working with my model, I found this to be difficult. Because of the complexity of the model, it is very difficult to use the built-in modeling software, but at the same time, when I import the STP file into Trelis/gmsh, I will encounter some compatibility problems, such as loss of block information. Do you have any good suggestions for this problem? I look forward to your reply.
Fendy

**Reply by GiudGiud** [reply]
what do you currently do with the STP file? Does moose read those?
How did you generate your mesh?

Which simulation did converge? The one with the coarse mesh? how coarse was it?
Are you watching the CFL criterion?

Can you try to add a -pc_factor_shift NONZERO to your petsc options?

**Comment by GiudGiud** [comment]
Any update on this?

**Reply by Submarinewake** [reply]
Sorry, thank you for your attention. In response to this problem, after I further tried the above solutions, the problem turned into a difficulty in convergence. Since I was trying other work recently, I did not continue to discuss it in depth. I hope that I can find the problem and solve it later.
